checkout-example-advanced/Controllers/WebhookController.cs

```diff
@@ -0,0 +1,81 @@
+using Adyen.Model.Notification;
+using Adyen.Util;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+using adyen_dotnet_checkout_example_advanced.Options;
+
+namespace adyen_dotnet_checkout_example_advanced.Controllers
+{
+    [ApiController]
+    public class WebhookController : ControllerBase
+    {
+        private readonly ILogger<WebhookController> _logger;
+        private readonly HmacValidator _hmacValidator;
+        private readonly string _hmacKey;
+        
+        public WebhookController(ILogger<WebhookController> logger, IOptions<AdyenOptions> options, HmacValidator hmacValidator)
+        {
+            _logger = logger;
+            _hmacKey = options.Value.ADYEN_HMAC_KEY;
+            _hmacValidator = hmacValidator;
+        }
+
+        [HttpPost("api/webhooks/notifications")]
+        public async Task<ActionResult<string>> Webhooks(NotificationRequest notificationRequest)
+        {
+            _logger.LogInformation($"Webhook received::\n{notificationRequest}\n");
+
+            try
+            {
+                // JSON and HTTP POST notifications always contain a single `NotificationRequestItem` object.
+                // Read more: https://docs.adyen.com/development-resources/webhooks/understand-notifications#notification-structure.
+                NotificationRequestItemContainer container = notificationRequest.NotificationItemContainers?.FirstOrDefault();
+
+                if (container == null)
+                {
+                    return BadRequest("Container has no notification items.");
+                }
+
+                // We always recommend to activate HMAC validation in the webhooks for security reasons.
+                // Read more here: https://docs.adyen.com/development-resources/webhooks/verify-hmac-signatures & https://docs.adyen.com/development-resources/webhooks#accept-notifications.
+                if (!_hmacValidator.IsValidHmac(container.NotificationItem, _hmacKey))
+                {
+                    _logger.LogError($"Error while validating HMAC Key");
+                    return BadRequest("[not accepted invalid hmac key]");
+                }
+
+                // Return if webhook is not successful.
+                if (!container.NotificationItem.Success)
+                {
+                    _logger.LogError($"Webhook unsuccessful: {container.NotificationItem.Reason} \n" +
+                        $"EventCode: {container.NotificationItem.EventCode}");
+                    return Ok("[accepted]"); // The webhook was delivered (but was unsuccessful), hence why we'll return a [accepted] response to confirm that we've received it.
```

Comment:
The `[accepted`] response is sent in any case once the notification is processed, so here I think it is enough to log the status. In theory a merchant would probably want to process the `success:false` event anyway to update their backend

Reply:
Good note. I gave it a second thought, I've modified all WebhookControllers in the demos to include the Success/Unsuccessful webhook scenarios. I've moved the logic from top-level to the notification-level.

It will:
1) Check the EventCode
2) Check the Success flag
3) Do processing regardless of success:true / success:false (as an example, we just log the `Reason` when it's unsuccessful)

Please have a look at this in `WebhookController.cs` foreach of the samples @gcatanese